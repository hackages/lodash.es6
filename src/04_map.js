/**
 * Creates a new object where every value is mapped using callback
 */
export function map(collection, callback) {}
