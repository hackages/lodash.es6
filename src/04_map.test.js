import {map} from './04_map'

describe('map', () => {
  it('it should work with the identity function', () => {
    const results = map([1, 2, 3, 4], x => x)

    expect(results).toMatchSnapshot()
  })

  it('it should return the index', () => {
    const results = map([1, 1, 1], (x, index) => index)

    expect(results).toMatchSnapshot()
  })

  it('it should return the entire array as the third parameter', () => {
    const results = map([1, 2], (a, b, arr) => arr)

    expect(results).toMatchSnapshot()
  })
})
