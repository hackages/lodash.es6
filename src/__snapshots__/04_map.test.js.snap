// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map it should return the entire array as the third parameter 1`] = `
Array [
  Array [
    1,
    2,
  ],
  Array [
    1,
    2,
  ],
]
`;

exports[`map it should return the index 1`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`map it should work with the identity function 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;
